code/__HELPERS/text.dm

```diff
@@ -450,6 +450,17 @@ proc/checkhtml(var/t)
 			return 1
 	return 0
 
+//finds the first occurrence of one of the characters from needles argument inside haystack
+//it may appear this can be optimised, but it really can't. findtext() is so much faster than anything you can do in byondcode.

```

Comment:
findtext is actually a piece of shit so I have no idea what this author is talking about lol
NTSL benefited by like a factor of 5 from converting findtext calls to a single split call followed by array indexes
Don't bother changing this though it's not worth it